product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2015 OdooMRP team
+# Copyright 2015 AvanzOSC
+# Copyright 2015 Tecnativa
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+import datetime
+
+from odoo import api, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def name_get(self):
+        res = super(ProductProduct, self.with_context(customerinfo=True)).name_get()
+        return res
+
+    @api.model
+    def _name_search(
+        self, name="", args=None, operator="ilike", limit=100, name_get_uid=None
+    ):
+        res = super(ProductProduct, self)._name_search(
+            name, args=args, operator=operator, limit=limit, name_get_uid=name_get_uid
+        )
+        res_ids = list(res)
+        res_ids_len = len(res_ids)
+        if not limit or res_ids_len >= limit:
+            limit = (limit - res_ids_len) if limit else False
+        if (
+            not name
+            and limit
+            or not self._context.get("partner_id")
+            or res_ids_len >= limit
+        ):
+            return res_ids
+        limit -= res_ids_len
+        customerinfo_ids = self.env["product.customerinfo"]._search(
+            [
+                ("name", "=", self._context.get("partner_id")),
```

Comment:
```suggestion
                ("partner_id", "=", self._context.get("partner_id")),
```
Please, remember that.

Reply:
Ok, thanks.